R/get_source_expressions.R

```diff
@@ -139,6 +139,41 @@ get_source_expressions <- function(filename, lines = NULL) {
         )
       }
 
+      parse_error_rx <- rex(
+        start,
+        capture(anything, name = "msg_1"),
+        or(" at ", " on ", " ("),
+        "line ",
+        capture(digits, name = "line"),
+        maybe(")"),
+        capture(anything, name = "msg_2"),
+        end
+      )
+
+      if (grepl(parse_error_rx, e$message, perl = TRUE)) {
+        rx_match <- re_matches(
+          e$message,
+          parse_error_rx
+        )
+        l <- as.integer(rx_match$line)
+        # Sometimes the parser "line" runs one past the last line
+        l <- pmin(l, length(source_file$lines))
```

Comment:
will Lint break if we return say Inf here? the last line may be misleading if the issue was an unmatched brace much earlier...

maybe an issue for a future PR, feel free to add a TODO to investigate

Reply:
Pretty sure this breaks all lintr integrations that use source locations (e.g. RStudio Markers and GitHub linting).
I think this was the reason we even discovered the bug.